src/Devices/GateReader/GateReaderDevice.cs

```diff
@@ -30,28 +30,49 @@ namespace Transportation.Demo.Devices.GateReader
 {
     public class GateReaderDevice : BaseDevice
     {
-        public enum GateDirection { In, Out };
-
         private GateDirection CurrentDirection;
         private GateReaderDeviceConfig deviceConfig;
 
+        /// <summary>
+        /// Device class constructor, calls base constructor and starts loading device values. 
+        /// Accepts a device config, client, and event scheduler. The later 2 are externalized to make
+        /// unit testing of the device easier. 
+        /// </summary>
         public GateReaderDevice(GateReaderDeviceConfig deviceConfig, IDeviceClient client, IEventScheduler eventScheduler) 
             : base(deviceConfig, client, eventScheduler)
         {
             // save device configuration
             this.deviceConfig = deviceConfig;
+        }
 
+        public new Task InitializeAsync()
+        {
             // set initial direction
-            this.SetDirectionAsync((GateDirection)Enum.Parse(typeof(GateDirection), deviceConfig.initialDirection)).Wait();
+            string initialDirection = deviceConfig.initialDirection;            // use configuration value as default
+            var myTwin = _DeviceClient.GetDynamicDigitalTwinAsync().Result;            // get twin 
+            if (myTwin.Properties.Reported.Contains("GateDirection"))
```

Comment:
cast myTwin as `Twin` first, and validate it for null before trying to access `.Properties` on it.